### ISSUE TITLE
Mouse focus not moved to viewer when 'activate by hover' is set in Windows10

### ISSUE THREAD

**Issue by sigwx** [TurboVNC/turbovnc#163]
When enabling 'Activate a window by hovering over it with the mouse' in Windows 10, when I hover over the TurboVNC Viewer, focus stays with the last windows application rather than switching to TurboVNC.  Works for all the other applications on the desktop, so thought it might be something specific to TurboVNC.    Currently using the TurboVNC-2.2.80-x64.exe from Feb 5.

**Comment by dcommander** [comment]
It's more likely something specific to Java.

**Comment by sigwx** [comment]
I would agree with you, however the older viewers didn't seem to display this issue.  Or has the viewer implementation on Windows changed?

**Comment by dcommander** [comment]
It depends on what you mean by "the older viewers."  In TurboVNC 2.2.x and earlier, there were two Windows viewers: the Windows native viewer, which has been part of TurboVNC since the beginning (based originally on TightVNC 1.3.x but with numerous enhancements over its 15-year history), and the Windows/Java viewer, which was originally only for browser-based use cases but which matured to the point that it could be used as a standalone viewer.  In TurboVNC 3.0 (2.2.80 is a 3.0 alpha/evolving build), the Windows native viewer has been retired, and the only Windows viewer is now the Java viewer with an embedded JRE based on OpenJDK 11.

**Comment by dcommander** [comment]
A useful data point would be testing the TurboVNC 2.2.1 release and confirming whether the issue also exists in the Windows/Java viewer that is part of that release.  That would tell me whether the issue is likely a Java thing or whether it's simply a regression.

**Comment by dcommander** [comment]
Suspecting that it's a Java thing:
https://netbeans.org/bugzilla/show_bug.cgi?id=230805

**Comment by sigwx** [comment]
Its probably a Java thing.  I didn't realize that the underlying implementation had changed (I was running a 2.2.80 from last fall so that was probably the native version.  I've seen similar Java issues in the past,  though I was hoping by now they would have dealt with it.  Thanks for the info!

**Comment by dcommander** [comment]
I will investigate whether there's any way to work around it.

**Comment by sigwx** [comment]
Another data point... when I move the mouse over the viewer, the mouse events are making it over (the titlebars of the windows in X light up when the mouse is over them, the cursor in a terminal switches from hollow to filled) its just that the keyboard focus in Windows isn't moving to the viewer (if that makes any sense).
